### DISCUSSION TITLE
Build fails with invalid covariant return type, 15.1.1

### DISCUSSION THREAD

**Discussion by jrobcary** [trilinos/Trilinos#12918] [Q&A] [answered]
/home/research/cary/projects/xsimall-dev1/builds/trilinos-15.1.1/packages/muelu/src/../adapters/stratimikos/Thyra_MueLuPreconditionerFactory_def.hpp:408:6:   required from here
/home/research/cary/projects/xsimall-dev1/builds/trilinos-15.1.1/packages/xpetra/src/Operator/Xpetra_TpetraHalfPrecisionOperator.hpp:130:70: error: invalid covariant return type for ‘Teuchos::RCP<const Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > Xpetra::TpetraHalfPrecisionOperator<Scalar, LocalOrdinal, GlobalOrdinal, Node>::getDomainMap() const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = long long int; Node = Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial>]’

the configuration is

'/home/research/cary/opt/dev1/contrib-gcc1120/cmake-3.26.4-ser/bin/cmake' \
  -DCMAKE_INSTALL_PREFIX:PATH=/home/research/cary/opt/dev1/contrib-gcc1120/trilinos-15.1.1-ser \
  -DCMAKE_BUILD_TYPE:STRING=Release \
  -DCMAKE_COLOR_MAKEFILE:BOOL=FALSE \
  -DCMAKE_VERBOSE_MAKEFILE:BOOL=TRUE \
  -DCMAKE_C_COMPILER:FILEPATH='/home/bilder/contrib/gcc-11.2.0/bin/gcc' \
  -DCMAKE_CXX_COMPILER:FILEPATH='/home/bilder/contrib/gcc-11.2.0/bin/g++' \
  -DCMAKE_Fortran_COMPILER:FILEPATH='/home/bilder/contrib/gcc-11.2.0/bin/gfortran' \
  -DCMAKE_Fortran_FLAGS:STRING='-fallow-argument-mismatch -pipe -fPIC' \
  -DCMAKE_C_FLAGS:STRING='-pthread -pipe -fPIC' \
  -DCMAKE_CXX_FLAGS:STRING='-pthread -Wno-deprecated-declarations -pipe -fPIC -fopenmp' \
  -DDART_TESTING_TIMEOUT:STRING=600 \
  -DML_ENABLE_SuperLU:BOOL=OFF \
  -DKokkosKernels_ENABLE_EXPERIMENTAL:BOOL=OFF \
  -DTrilinos_ENABLE_SuperLU5_API=ON \
  -DPYTHON_EXECUTABLE:FILEPATH=/home/research/cary/opt/dev1/contrib-gcc1120/Python-3.10.4-sersh/bin/python3 \
  -DTrilinos_ENABLE_SERIAL:BOOL='ON' \
  -DTpetra_ENABLE_SERIAL:BOOL='ON' \
  -DKokkos_ENABLE_SERIAL:BOOL='ON' \
  -DTrilinos_INST_SERIAL:BOOL='ON' \
  -DTpetra_INST_SERIAL:BOOL='ON' \
  -DKokkos_INST_SERIAL:BOOL='ON' \
  -DTrilinos_ENABLE_EXAMPLES:BOOL=FALSE \
  -DTrilinos_ENABLE_FLOAT:BOOL=ON \
  -DTpetra_INST_INT_LONG_LONG:BOOL=ON \
  -DCMAKE_EXE_LINKER_FLAGS:STRING='-L/home/bilder/contrib/gcc-11.2.0/lib64 -Wl,-rpath,/home/bilder/contrib/gcc-11.2.0/lib64' \
  -DTPL_BLAS_LIBRARIES:FILEPATH='/home/research/cary/opt/dev1/contrib-gcc1120/lapack-3.7.0-ser/lib64/libblas.a' \
  -DTPL_LAPACK_LIBRARIES:FILEPATH='/home/research/cary/opt/dev1/contrib-gcc1120/lapack-3.7.0-ser/lib64/liblapack.a' \
  -DTrilinos_ENABLE_Teuchos:BOOL=ON \
  -DTrilinos_ENABLE_Triutils:BOOL=ON \
  -DTrilinos_ENABLE_Epetra:BOOL=ON \
  -DTrilinos_ENABLE_Amesos:BOOL=ON \
  -DTrilinos_ENABLE_Ifpack:BOOL=ON \
  -DTrilinos_ENABLE_AztecOO:BOOL=ON \
  -DTrilinos_ENABLE_ML:BOOL=ON \
  -DTrilinos_ENABLE_EpetraExt:BOOL=ON \
  -DTrilinos_ENABLE_Galeri:BOOL=ON \
  -DTrilinos_ENABLE_Shards:BOOL=ON \
  -DTrilinos_ENABLE_Intrepid:BOOL=ON \
  -DTrilinos_ENABLE_Komplex:BOOL=ON \
  -DTrilinos_ENABLE_NOX:BOOL=ON \
  -DTrilinos_ENABLE_Kokkos:BOOL=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_Xpetra:BOOL=ON \
  -DTrilinos_ENABLE_MueLu:BOOL=ON \
  -DTrilinos_ENABLE_Amesos2:BOOL=ON \
  -DTrilinos_ENABLE_Belos:BOOL=ON \
  -DTrilinos_ENABLE_Ifpack2:BOOL=ON \
  -DTPL_ENABLE_BinUtils:BOOL=OFF \
  -DTrilinos_ENABLE_Zoltan:BOOL=OFF \
  -DTPL_ENABLE_SuperLU:BOOL=ON \
  -DSuperLU_INCLUDE_DIRS:PATH='/home/research/cary/opt/dev1/contrib-gcc1120/superlu-6.0.1-ser/include' \
  -DSuperLU_LIBRARY_DIRS:PATH='/home/research/cary/opt/dev1/contrib-gcc1120/superlu-6.0.1-ser/lib64' \
  -DSuperLU_LIBRARY_NAMES:STRING='superlu' \
  -DTPL_ENABLE_HDF5:BOOL=ON \
  -DHDF5_INCLUDE_DIRS:PATH='/home/research/cary/opt/dev1/contrib-gcc1120/hdf5-1.12.3-ser/include' \
  -DHDF5_LIBRARY_DIRS:PATH='/home/research/cary/opt/dev1/contrib-gcc1120/hdf5-1.12.3-ser/lib' \
  -DHDF5_LIBRARY_NAMES:STRING='hdf5_hl;hdf5;hdf5_tools;z;dl' /home/research/cary/projects/xsimall-dev1/builds/trilinos-15.1.1

**Comment by cgcgcg** [comment, marked as answer]
Maybe fixed by #12819.

**Comment by jrobcary** [comment]
thx.
